data/onPostBuild/generateMarkdown.ts

```diff
@@ -0,0 +1,210 @@
+import fastGlob from 'fast-glob';
+
+/**
+ * This script generates Markdown static files alongside HTML files for each page.
+ * This allows LLM crawlers and other clients to request markdown versions of pages
+ * which are significantly more token-efficient than HTML.
+ */
+
+const REPORTER_PREFIX = 'generateMarkdown:';
+
+// Constants for content validation
+const REDIRECT_PAGE_MAX_SIZE = 1000; // Maximum size in bytes for redirect pages
+const MIN_CONTENT_LENGTH = 100; // Minimum content length to consider meaningful
+
+// Configure Turndown for documentation-friendly markdown
+const createTurndownService = () => {
+  const turndownService = new TurndownService({
+    headingStyle: 'atx',
+    codeBlockStyle: 'fenced',
+    bulletListMarker: '-',
+    emDelimiter: '_',
+    strongDelimiter: '**',
+    linkStyle: 'inlined',
+    linkReferenceStyle: 'full',
+  });
+
+  // Preserve code block language annotations
+  turndownService.addRule('fencedCodeBlock', {
+    filter: (node) => {
+      return node.nodeName === 'PRE' && node.firstChild && node.firstChild.nodeName === 'CODE';
+    },
+    replacement: (content, node) => {
+      const codeNode = node.firstChild as HTMLElement;
+      const className = codeNode.getAttribute('class') || '';
+      const languageMatch = className.match(/language-(\w+)/);
+      const language = languageMatch ? languageMatch[1] : '';
+
+      // Get the actual code content
+      const code = codeNode.textContent || '';
+
+      return '\n\n```' + language + '\n' + code + '\n```\n\n';
+    },
+  });
+
+  // Remove navigation, headers, footers, and other UI elements
+  turndownService.remove(['nav', 'header', 'footer', 'script', 'style', 'noscript']);
+
+  return turndownService;
+};
+
+// Extract main article content from HTML file
+const extractMainContent = (htmlPath: string): string | null => {
+  try {
+    if (!fs.existsSync(htmlPath)) {
+      return null;
+    }
+
+    const html = fs.readFileSync(htmlPath, 'utf8');
+
+    // Check if this is a redirect page (very small file with window.location.href)
+    if (html.length < REDIRECT_PAGE_MAX_SIZE && html.includes('window.location.href')) {
+      return null; // Skip redirect pages
+    }
+
+    const $ = cheerio.load(html);
+
+    // Remove unwanted elements
+    $('nav, header, footer, script, style, noscript, .sidebar, .navigation').remove();
+
+    // Try to find the main article content
+    // Look for common article containers
+    let mainContent = $('article').html() || $('main').html() || $('#main-content').html();
+
+    // If we can't find a main content area, fall back to body but remove header/footer
+    if (!mainContent) {
+      $('body > header, body > footer, body > nav').remove();
+      mainContent = $('body').html();
+    }
+
+    // Check if content is meaningful (more than just whitespace/empty tags)
+    if (mainContent && mainContent.trim().length < MIN_CONTENT_LENGTH) {
+      return null; // Skip pages with minimal content
```

Comment:
If we skip stuff for certain pages, how do we handle this in the frontend? The Open in Markdown button is available all the time, but the pages may not be